addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -215,15 +215,28 @@ function factory(dependencies) {
             } else if (!this.mimetype) {
                 return clear();
             }
+            switch (this.mimetype) {
+                case 'image/bmp':
+                case 'image/gif':
+                case 'image/jpeg':
+                case 'image/png':
+                case 'image/webp':
+                    return 'image';
+                case 'text/css':
+                case 'text/html':
+                case 'text/plain':
+                    return 'text';
+                case 'video/mp4':
+                case 'video/ogg':
+                case 'video/webm':
+                    return 'video';
+            }
             const match = this.type === 'url'
                 ? this.url.match('(youtu|.png|.jpg|.gif)')
-                : this.mimetype.match('(image|video|application/pdf|text)');
+                : this.mimetype.match('(application/pdf)');
             if (!match) {
                 return clear();
             }
-            if (match[1].match('(.png|.jpg|.gif)')) {
-                return 'image';
-            }
```

Comment:
The return value below becomes not correct if you remove this, `match[1]` is the extension not the string version the method is supposed to return.

Reply:
But the png/jpg/gif are handled in the switch case above. That's why this has been removed.

Reply:
Oh okay I see, url with them should also have the appropriate file type... I'll handle them accordingly